### PR TITLE
Implement Stock Reconciliation Engine

### PR DESCRIPTION
**Pull request by mrtripop**
### Summary

  Implemented an automated system to reconcile the redundant Batch.quantity field with the actual sum of StoreStock quantities to prevent data drift.

  - Added sumQuantityByBatchId to StoreStockRepository for accurate stock aggregation.
  - Implemented StockReconciliationService with paginated processing of all batches.
  - Integrated optimistic locking handling to ensure resilience during concurrent updates.
  - Added a daily cron job at 2 AM to automate reconciliation.
  - Provided an administrative API endpoint to manually trigger the reconciliation process.
  - Comprehensive TDD suite covering business success, edge cases (synced/orphaned), and technical resilience.

  Test Plan

  - [ ] Run ./mvnw test -Dtest=StockReconciliationServiceTest to verify core logic, edge cases, and concurrency handling.
  - [ ] Verify StockReconciliationScheduler is correctly annotated for the 2 AM daily run.
  - [ ] Verify InventoryAdminController is protected by hasRole('ADMIN').

### CODE REVIEW

File: src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  private final BatchRepository batchRepository;
+  private final StoreStockRepository storeStockRepository;
+  private final AuditService auditService;
+
+  @Override
+  public void reconcileAll() {
+    int pageNumber = 0;
+    Page<Batch> batchPage;
+    do {
+      batchPage = batchRepository.findAll(PageRequest.of(pageNumber++, 100, Sort.by("id")));
```
Comment (mrtripop) [inline src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java:33]: Required: Hardcode value, use configuration instead.

File: src/main/java/com/mrtripop/inventory/scheduler/StockReconciliationScheduler.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class StockReconciliationScheduler {
+    private final StockReconciliationService reconciliationService;
+
+    @Scheduled(cron = "0 0 2 * * ?") // 2 AM daily
```
Comment (mrtripop) [inline src/main/java/com/mrtripop/inventory/scheduler/StockReconciliationScheduler.java:15]: Required: Hardcode corn job, use configuration instead.

File: src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      for (Batch batch : batchPage) {
+        try {
+          reconcileBatch(batch.getId());
+        } catch (Exception e) {
+          log.error("Failed to reconcile batch {}: ", batch.getId(), e);
+        }
+      }
+    } while (batchPage.hasNext());
```
Comment (mrtripop) [inline src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java:41]: Suggest: Maybe refactor using Java Stream API for easier to read and maintain, but depend on your consideration.

File: src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      }
+
+      auditService.recordAudit(
+          StockReconciliationService.ACTION_RECONCILIATION,
+          StockReconciliationService.ENTITY_BATCH,
+          batch.getId().toString(),
+          String.valueOf(oldQuantity),
+          String.valueOf(actualSum)
+      );
+    }
```
Comment (mrtripop) [inline src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java:70]: Required: Use if-cause instead of nested-if

File: src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    long oldQuantity = batch.getQuantity();
+
+    if (oldQuantity != actualSum) {
+      log.info("Corrected stock drift for batch {}: {} -> {}", batchId, oldQuantity, actualSum);
+      batch.setQuantity(actualSum);
+      try {
+        batchRepository.save(batch);
+      } catch (ObjectOptimisticLockingFailureException e) {
+        log.warn("Optimistic lock failure reconciling batch {}: skipping", batchId);
+        return;
```
Comment (mrtripop) [inline src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java:60]: Suggest: In the future, add new capability to handle race condition gracefully than this.

File: src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java (hunk 1 above)
Comment (mrtripop) [inline src/main/java/com/mrtripop/inventory/services/impl/StockReconciliationServiceImpl.java:33]: Suggest: I think count total and calculate all task, then create pending task for ability to parallel process with other process.

File: src/main/java/com/mrtripop/inventory/scheduler/StockReconciliationScheduler.java (hunk 2 above)
Comment (mrtripop) [reply src/main/java/com/mrtripop/inventory/scheduler/StockReconciliationScheduler.java:15]: Add guide to prevent AI to not hardcode

File: src/test/java/com/mrtripop/inventory/services/StockReconciliationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      Batch batch = defaultBatch();
+      when(batchRepository.findById(VALID_BATCH_ID)).thenReturn(Optional.of(batch));
+      when(storeStockRepository.sumQuantityByBatchId(VALID_BATCH_ID)).thenReturn(INITIAL_BATCH_QTY);
+
+      // Act
+      service.reconcileBatch(VALID_BATCH_ID);
+
+      // Assert
+      verify(batchRepository, never()).save(any());
+      verify(auditService, never()).recordAudit(any(), any(), any(), any(), any());
```
Comment (mrtripop) [inline src/test/java/com/mrtripop/inventory/services/StockReconciliationServiceTest.java:82]: Mutation testing may failed because it never catch bug in test like this
